webapp/src/js/services/db-sync.js

```diff
@@ -52,122 +46,110 @@ angular
       );
     };
 
-    var getOptions = function(direction) {
-      var options = {};
-      if (direction === 'to') {
-        options.checkpoint = 'source';
-        options.filter = readOnlyFilter;
+    const DIRECTIONS = [
+      {
+        name: 'to',
+        options: {
+          checkpoint: 'source',
+          filter: readOnlyFilter
+        },
+        allowed: () => Auth('can_edit').then(() => true).catch(() => false)
+      },
+      {
+        name: 'from',
+        options: {},
+        allowed: () => $q.resolve(true)
       }
-
-      return options;
-    };
-
-    var replicate = function(direction) {
-      var options = getOptions(direction);
-      var remote = DB({ remote: true });
-      return DB()
-        .replicate[direction](remote, options)
-        .on('denied', function(err) {
-          // In theory this could be caused by 401s
-          // TODO: work out what `err` looks like and navigate to login
-          // when we detect it's a 401
-          $log.error('Denied replicating ' + direction + ' remote server', err);
-        })
-        .on('error', function(err) {
-          $log.error('Error replicating ' + direction + ' remote server', err);
-        })
-        .on('complete', function(info) {
-          if (!info.ok && authenticationIssue(info.errors)) {
-            Session.navigateToLogin();
-          }
-        });
-    };
-
-    const replicateTo = () => {
-      const AUTH_FAILURE = {};
-      return Auth('can_edit')
-        // not authorized to replicate to server - that's ok. Silently skip replication.to
-        .catch(() => AUTH_FAILURE)
-        .then(err => {
-          if (err !== AUTH_FAILURE) {
-            return replicate('to');
+    ];
+
+    const replicate = function(direction) {
+      return direction.allowed()
+        .then(allowed => {
+          if (!allowed) {
+            // not authorized to replicate - that's ok, skip silently
+            return;
           }
+          const remote = DB({ remote: true });
+          return DB()
+            .replicate[direction.name](remote, direction.options)
+            .on('denied', function(err) {
+              // In theory this could be caused by 401s
+              // TODO: work out what `err` looks like and navigate to login
+              // when we detect it's a 401
+              $log.error(`Denied replicating ${direction.name} remote server`, err);
+            })
+            .on('error', function(err) {
+              $log.error(`Error replicating ${direction.name} remote server`, err);
+            })
+            .on('complete', function(info) {
+              const authError = !info.ok &&
+                info.errors &&
+                info.errors.some(error => error.status === 401);
+              if (authError) {
```

Comment:
I think this is covered by the `.on('error')` handler above. We should probably add some better logic here because at the moment we get too much logging when replication fails when you're offline. I'll have a look in a separate PR...